packages/typespec-vscode/src/code-action-provider.ts

```diff
@@ -47,9 +49,87 @@ export class TypeSpecCodeActionProvider implements vscode.CodeActionProvider {
         );
       }
     });
+
+    actions.push(...(await this.createInstallPackageByNpm(context, actions)));
+
     return actions;
   }
 
+  private async createInstallPackageByNpm(
+    context: vscode.CodeActionContext,
+    actions: vscode.CodeAction[],
+  ): Promise<vscode.CodeAction[]> {
+    // When the corresponding node dependency package is not installed,
+    // consider generating a quick fix to install via npm command
+    const diagnostics = context.diagnostics.filter(
+      (diagnostic) =>
+        diagnostic.source === "TypeSpec" &&
+        diagnostic.code &&
+        diagnostic.code === "import-not-found",
+    );
+    if (diagnostics.length === 0) {
+      return [];
+    }
+
+    for (const diagnostic of diagnostics) {
+      if ("data" in diagnostic) {
+        if (diagnostic.data && typeof diagnostic.data === "object" && "file" in diagnostic.data) {
+          const packageFile = diagnostic.data.file as string;
+          const { packageJsonFolder, packageJson } = await searchAndLoadPackageJson(
+            getDirectoryPath(packageFile),
+          );
+          if (packageJsonFolder === undefined) {
+            continue;
+          }
+
+          // The message content is Couldn't resolve import "@typespec/http".
+          // The compiler's diagnostics do not provide a specific attribute to obtain the package name,
+          // so a regular expression is used to extract the package name within the double quotes
+          const targets = diagnostic.message.match(/"([^"]+)"/);
+          if (targets === null) {
+            continue;
+          }
+
+          // The position with index 0 contains double quotes, and the position with 1 does not contain
+          const packageNameIndex = 1;
+          const targetPackage = targets[packageNameIndex];
+
+          if (
```

Comment:
same as above. even the missing package does not set in the package.json, we still need to install.